### PR TITLE
Add Joystick support

### PR DESCRIPTION
**Pull request by murkle**
https://github.com/mattgodbolt/jsbeeb/issues/128

Buttons hard-coded to 10/11 which are the ones on the stick. maybe 0 and 1 would make more sense 🤷 

https://gamepad-tester.com/ is very cool for testing 😄

### CODE REVIEW

**Comment by murkle** [comment]
https://test.geogebra.org/~mike/jsBeebJoysticks/?autorun&loadBasic=https://test.geogebra.org/~mike/jsbeeb/joystickTest.bbc

**Comment by murkle** [comment]
/home/travis/build/mattgodbolt/jsbeeb/via.js:578
			var pads = navigator.getGamepads && navigator.getGamepads();
			           ^
ReferenceError: navigator is not defined

**Comment by murkle** [comment]
@mattgodbolt not sure what's wrong with that, seems OK in https://github.com/mattgodbolt/jsbeeb/blob/master/gamepads.js

**Comment by mattgodbolt** [comment]
Hi @murkle ! Thanks for this. The error is because the code is being run under unit test outside of a browser. I think changing the line to `navigator &&  navigator.getGamepads && navigator.getGamepads()` will make things happy. Will take a look at the rest of the code now :)

File: adc.js [hunk 1]
```diff
@@ -33,13 +33,49 @@ define([], function () {
     };
 
     Adc.prototype.onComplete = function () {
-        var val = 0x8000;
-        // TODO: switch on bottom two bits of adc_status and pick a value corresponding
-        // to the appropriate axis.
+      var val = 0x8000;
+		//val = Math.floor(Math.random() * 10000);
```
Comment (mattgodbolt) [inline adc.js]: Can you remove this comment, and indent the rest of this function consistently please? The rest of the codebase is using 4-space indents, not tabs. Cheers :)

File: via.js [hunk 2]
```diff
@@ -269,6 +269,24 @@ define(['./utils'], function (utils) {
                             temp &= 0x7f;
                             temp |= (self.t1_pb7 << 7);
                         }
+                        
+                        var buttons = this.getJoysticks();
+
+                        // clear PB4 and PB5
+                        temp = temp & 0xCF // 11001111
```
Comment (mattgodbolt) [inline via.js]: There's a number of missing semicolons: the `lint` check is picking these up too!

File: via.js [hunk 3]
```diff
@@ -269,6 +269,24 @@ define(['./utils'], function (utils) {
...
+
+                        // AUG p418
```
Comment (mattgodbolt) [inline via.js:278]: Thanks for the comment here: makes things a _lot_ clearer!

**Review by mattgodbolt — COMMENTED** [review]
Looks great! Minor formatting comments only -- and sorry for the silly test system not having a `navigator` defined - I hope the workaround is simple enough.

**Comment by murkle** [comment]
@mattgodbolt how's that now?

https://test.geogebra.org/~mike/jsBeebJoysticks/?autorun&loadBasic=https://test.geogebra.org/~mike/jsbeeb/joystickTest.bbc

**Comment by mattgodbolt** [comment]
Hi @murkle -tests are still failing. If you didn't know you can run `make lint` to get the same results locally :)

**Comment by mattgodbolt** [comment]
Oh darn it: I gave you bad info:

```
/home/travis/build/mattgodbolt/jsbeeb/via.js:578
            var pads = navigator && navigator.getGamepads && navigator.getGamepads();
```

is still failing :( Sorry!!! I'll look into how to get that working!

**Comment by mattgodbolt** [comment]
`make lint` is showing:

```
./via.js: line 591, col 10, Missing semicolon.
./via.js: line 603, col 10, Missing semicolon.
./adc.js: line 72, col 17, 'val' is already defined.
```


and `make test` would find the `navigator` issue locally if you haven't run it already.

**Comment by mattgodbolt** [comment]
If you fix up the lint issues, I'll merge and fix up the `navigator` thing. it might be as simple as hacking a magic parameter to `via` to pass the "navigator" in, or use a fake via in tests... but it can be my problem :)

**Comment by murkle** [comment]
Thanks, fixed those. navigator seems fine in gamepad.js, very odd :)

**Comment by mattgodbolt** [comment]
Cool. Yeah the gamepad.js isn't called by the "normal" emulation path in the tests. The VIA is though. Will fix :) Thanks!

**Review by mattgodbolt — APPROVED** [review]
thanks again!

**Pull request merged**